internal/worker/modelworkermanager/modelworkermanager.go

```diff
@@ -122,8 +116,8 @@ func (config Config) Validate() error {
 	if config.Logger == nil {
 		return errors.NotValidf("nil Logger")
 	}
-	if config.ModelWatcher == nil {
-		return errors.NotValidf("nil ModelConfigWatcher")
+	if config.ControllerDomainServices == nil {
+		return errors.NotValidf("nil ControllerDomainServices")
```

Comment:
Maybe `ControllerDomainServices Config`?

Reply:
Hi @leyao-daily , thanks for the review! For this I feel ControllerDomainServices would be better since it corresponds with the services.ControllerDomainServices, and it standardises with the rest of the checks in this block. We have another field called GetControllerConfig in this config, but the func is actually getting the config of the controller. In this case, we are only referencing the service in the controller domain. Perhaps @wallyworld could help to look at this too? Happy to change it if the both of you feel the same way :D

Reply:
It makes sense. I noticed that the origin code return `ModelConfigWatcher` for `config.ModelWatcher == nil`, which confused me for its inconsistent inside the block.

Reply:
Oh yup I think that might have be an accidental typo, because the rest of the block doesnt add a "Config"

Reply:
It makes sense.